src/core/webview/ClineProvider.ts

```diff
@@ -1080,6 +1080,41 @@ export class ClineProvider extends EventEmitter<ClineProviderEvents> implements
 					}
 					case "exportTaskWithId":
 						this.exportTaskWithId(message.text!)
+						break
+					case "importSettings":
+						const uris = await vscode.window.showOpenDialog({
+							filters: { JSON: ["json"] },
+							canSelectMany: false,
+						})
+
+						if (uris) {
+							if (message.text === "global") {
+								await this.contextProxy.importGlobalSettings(uris[0].fsPath)
+							} else {
+								await this.contextProxy.importGlobalSettings(uris[0].fsPath)
+							}
+
+							this.settingsImportedAt = Date.now()
+							await this.postStateToWebview()
+							await vscode.window.showInformationMessage(t("common:info.settings_imported"))
+						}
+						break
+					case "exportSettings":
+						const uri = await vscode.window.showSaveDialog({
+							filters: { JSON: ["json"] },
+							defaultUri: vscode.Uri.file(
+								path.join(os.homedir(), "Documents", `roo-code-${message.text}.json`),
```

Comment:
Does this default make sense in all OSes? What happens if we don't set a default?

Reply:
If you don't set a default then it defaults to the last directory you saved something to (if any), and the named is "Untitled.json".